src/main/java/net/sf/jabref/gui/keyboard/KeyBindingsDialogViewModel.java

```diff
@@ -0,0 +1,210 @@
+                            keyBinding.getCategory());
+                    TreeItem<KeyBindingWrapper> keyBindTreeItem = new TreeItem<>(keyBindWrapper);
+                    categoryItem.getChildren().add(keyBindTreeItem);
+                }
+            });
+            categoryItem.setExpanded(true);
+            root.getChildren().add(categoryItem);
+        }
+        root.setExpanded(true);
+        keyBindingsTable.setRoot(root);
+    }
+
+    private void bindColumnValues() {
+        actionColumn
+                .setCellValueFactory(cellData -> new SimpleStringProperty(cellData.getValue().getValue().getName()));
+        shortcutColumn.setCellValueFactory(cellData -> cellData.getValue().getValue().bindProperty());
+    }
+
+    private void registerKeyEvents() {
+        keyBindingsTable.setOnKeyPressed(evt -> grabKey(evt));
+    }
+
+    private Stage getDialogStage() {
+        return (Stage) closeButton.getScene().getWindow();
+    }
+
+    @FXML
+    private void closeDialog() {
+        Stage stage = getDialogStage();
+        stage.close();
+    }
+
+    @FXML
+    private void saveKeyBindingsAndCloseDialog() {
+        Globals.getKeyPrefs().setNewKeyBindings(keyBindingRepository.getKeyBindings());
+
+        // show success message
+        FXAlert alert = new FXAlert(AlertType.INFORMATION, Localization.lang("Key bindings changed"));
+        alert.setHeaderText(null);
+        alert.setContentText(Localization.lang("Your new key bindings have been stored.") + '\n'
+                + Localization.lang("You must restart JabRef for the new key bindings to work properly."));
+        alert.getButtonTypes().clear();
+        alert.getButtonTypes().add(ButtonType.CLOSE);
+        alert.showAndWait().ifPresent(response -> {
+            if (response == ButtonType.CLOSE) {
+                closeDialog();
+            }
+        });
+    }
+
+    @FXML
+    private void setDefaultBindings() {
+        ObservableList<TreeItem<KeyBindingWrapper>> selectedKeyBindingWrapperItems = keyBindingsTable
+                .getSelectionModel().getSelectedItems();
+        if (selectedKeyBindingWrapperItems.isEmpty()) {
+            FXAlert alert = new FXAlert(AlertType.INFORMATION, Localization.lang("Resetting all key bindings"));
+            alert.setHeaderText(null);
+            alert.setContentText(Localization.lang("All key bindings will be reset to their defaults."));
+            alert.getButtonTypes().clear();
+            ButtonType resetButtonType = new ButtonType("Reset", ButtonData.OK_DONE);
+            alert.getDialogPane().getButtonTypes().addAll(resetButtonType, ButtonType.CANCEL);
+            alert.showAndWait().ifPresent(response -> {
+                if (response == resetButtonType) {
+                    keyBindingRepository.resetToDefault();
+                    populateTable();
+                }
+            });
+        } else {
+            selectedKeyBindingWrapperItems.forEach(keyBindWrapperItem -> {
```

Comment:
I find it counterintuitive that the same button sometimes resets only a few bindings and sometimes all.
 Proposal: add a third column to the table which contains a small reset button (only icon, light gray by default, dark gray on hovering the row) which resets the corresponding binding. Then the button at the button is a `Reset all bindings` button.